packages/react/modules/components/configure-adapter/configure-adapter.js

```diff
@@ -27,19 +23,96 @@ type Props = {
   defaultFlags?: Flags,
   children: React.Component<any>,
 };
+type State = {
+  adapterArgs: AdapterArgs,
+};
 type AdapterState = $Values<typeof AdapterStates>;
+type AdapterReconfigurationOptions = {
+  exact?: boolean,
+};
+type AdapterReconfiguration = {
+  adapterArgs: AdapterArgs,
+  options: AdapterReconfigurationOptions,
+};
+type ReconfigureAdapter = (
+  adapterArgs: AdapterArgs,
+  options: AdapterReconfigurationOptions
+) => void;
+
+export const AdapterContext: Context<ReconfigureAdapter> = createReactContext(
+  () => {}
+);
 
-export default class ConfigureAdapter extends PureComponent<Props> {
+export const mergeAdapterArgs = (
+  previousAdapterArgs: AdapterArgs,
+  { adapterArgs: nextAdapterArgs, options }: AdapterReconfiguration
+): AdapterArgs =>
+  options.exact ? nextAdapterArgs : merge(previousAdapterArgs, nextAdapterArgs);
+
+export default class ConfigureAdapter extends PureComponent<Props, State> {
   static defaultProps = {
     shouldDeferAdapterConfiguration: false,
     children: null,
     defaultFlags: {},
   };
 
   adapterState: AdapterState = AdapterStates.UNCONFIGURED;
-  setAdapterState = (nextAdapterState: AdapterState): void => {
+  pendingAdapterArgs: ?AdapterArgs = null;
+
+  state: { adapterArgs: AdapterArgs } = {
+    adapterArgs: this.props.adapterArgs,
+  };
+
+  setAdapterState = (
```

Comment:
Nitpick: isn't this meant to be more like a `cacheAdapterState`, since it's stored in the component instance?

Reply:
Not sure. Maybe it's more `appliedAdapterArgs` as soon as they end up on component state they should be applied to the adapter. Will rename.